compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/CfaUtils.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.cfa
+
+import kotlinx.collections.immutable.PersistentMap
+import kotlinx.collections.immutable.persistentMapOf
+import org.jetbrains.kotlin.contracts.description.EventOccurrencesRange
+import org.jetbrains.kotlin.fir.references.FirResolvedNamedReference
+import org.jetbrains.kotlin.fir.resolve.dfa.cfg.*
+import org.jetbrains.kotlin.fir.symbols.impl.FirPropertySymbol
+
+
+class PropertyInitializationInfo(
+    map: PersistentMap<FirPropertySymbol, EventOccurrencesRange> = persistentMapOf()
+) : ControlFlowInfo<PropertyInitializationInfo, FirPropertySymbol, EventOccurrencesRange>(map) {
+    companion object {
+        val EMPTY = PropertyInitializationInfo()
+    }
+
+    override val constructor: (PersistentMap<FirPropertySymbol, EventOccurrencesRange>) -> PropertyInitializationInfo =
+        ::PropertyInitializationInfo
+
+    fun merge(other: PropertyInitializationInfo): PropertyInitializationInfo {
+        var result = this
+        for (symbol in keys.union(other.keys)) {
+            val kind1 = this[symbol] ?: EventOccurrencesRange.ZERO
+            val kind2 = other[symbol] ?: EventOccurrencesRange.ZERO
+            result = result.put(symbol, kind1 or kind2)
+        }
+        return result
+    }
+}
+
+class LocalPropertyCollector private constructor() : ControlFlowGraphVisitorVoid() {
+    companion object {
+        fun collect(graph: ControlFlowGraph): MutableSet<FirPropertySymbol> {
+            val collector = LocalPropertyCollector()
+            graph.traverse(TraverseDirection.Forward, collector)
+            return collector.symbols
+        }
+    }
+
+    private val symbols: MutableSet<FirPropertySymbol> = mutableSetOf()
+
+    override fun visitNode(node: CFGNode<*>) {}
+
+    override fun visitVariableDeclarationNode(node: VariableDeclarationNode) {
+        symbols += node.fir.symbol
+    }
+}
+
+class DataCollector(private val localProperties: Set<FirPropertySymbol>) :
```

Comment:
`DataCollector` is too common name. Please rename it to something like `PropertyInitializationInfoCollector`